src/range_proof/mod.rs

```diff
@@ -287,18 +287,19 @@ impl RangeProof {
         transcript.rangeproof_domain_sep(n as u64, m as u64);
 
         for V in value_commitments.iter() {
-            transcript.commit_point(b"V", V);
+            transcript.validate_and_commit_point(b"V", V)?;
```

Comment:
On a properly-generated Pedersen commitment, the blinding factor will be zero with negligible probability, so a zero blinding factor almost certainly indicates an RNG failure and I don't think there's a good reason to accept them.

It might be the case that some higher-level protocol uses inputs that aren't secret, as you mentioned (private communication), and so someone might choose a zero blinding factor. I'm not sure this is a good design compared to forming the commitment properly and giving the opening `(v, v_blinding)`; if the blinding factor is set to zero, the committer still has to supply `v` (since there's not an easy way to recover `v` from `vB`, even for low-entropy `v`).  So this "optimization" doesn't seem to really simplify the structure, the committer still has to supply the opening, it's just that the opening is smaller.

That said, I'm not totally set on forcing the `V` commitments to be nonzero, I just don't see a good reason to allow it, and there are (in my opinion) good reasons not to allow it (see next comment).

Reply:
I could see two options for resolving this:

1.  Remove the `V != 0` check to allow the zero-blinding-factors case, and leave other checks in place;
2.  Leave all of the checks in place, and later, if `V != 0` turns out to block something, we can relax it in the future (since relaxing a check is easier to do later than adding one).

WDYT?

Reply:
Lets do (1) and merge this PR?